### ISSUE TITLE
Execution failed for task ':stripe_stripe-react-native:compileDebugKotlin'.

### ISSUE THREAD

**Issue by nitinvarda** [stripe/stripe-react-native#1560]
**Describe the bug**
Initialized react-native project, Installed  @stripe/stripe-react-native and tried to run android, it throws an error

`
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':stripe_stripe-react-native:compileDebugKotlin'.
> Error while evaluating property 'compilerOptions.jvmTarget' of task ':stripe_stripe-react-native:compileDebugKotlin'.
   > Failed to calculate the value of property 'jvmTarget'.
      > Unknown Kotlin JVM target: 20
`

**Expected behavior**
The app builds successfully when using the library with fresh RN projects.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - Mac OS Sonoma  14.0 

**Smartphone (please complete the following information):**
 - Android emulator

**Additional context**
 - react: 18.2.0
 - react-native : 0.72.7

**Comment by hayate** [comment]
@nitinvarda had the same issue as you. Upgrading Kotlin version fixed the issue.

In the file `android/build.gradle` I had: `kotlinVersion = "1.7.0"` changing it to: `kotlinVersion = "1.8.0"`
then `./gradlew clean` and `./gradlew bundleRelease` and the build succeeded.

**Comment by nitinvarda** [comment]
@hayate great you found a solution. Can you please tell me where to change the kotlin version, is it in build.gradle file or do I have to do that on Android studio?

**Comment by nguyenhoanganhdev** [comment]
@hayate In my file android/build.gradle kotlinVersion doesn't have specify version. What should I do

targetSdkVersion = Integer.parseInt(findProperty('android.targetSdkVersion') ?: '33')
                if (findProperty('android.kotlinVersion')) {
            kotlinVersion = findProperty('android.kotlinVersion')
        }

**Comment by rakshitbharat** [comment]
same error

**Comment by rakshitbharat** [comment]
#551 
this will help

**Comment by Aryan45231** [comment]
* What went wrong:
Execution failed for task ':stripe_stripe-react-native:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 57s
info Run CLI with --verbose flag for more details.
facing same issue 
using reactnative : 0.73.0"
@stripe/stripe-react-native v 0.35.0.

**Comment by toscalivia83** [comment]
Having the same error

**Comment by ketankapale8** [comment]
same!!! Error!!!

**Comment by rishabraghav** [comment]
@Aryan45231 I'm getting same issue as your's how did you solve it ?

**Comment by nvdnvd00** [comment]
After upgrade to `react-native 0.73.2` with `kotlinVersion 1.8.0`, I'm facing the same issue
"@stripe/stripe-react-native": "^0.34.0"

**Comment by ratneshkumarroot** [comment]
same!!! Error!!!

**Comment by daniloeder** [comment]
I've solved this problem after removing this line from package.json dependencies:
"expo-application": "~5.9.1".

**Comment by rodrigo-nexudus** [comment]
I'm having this issue after updating to RN 0.75.2. Anyone have a fix? I've tried all the previously mentioned ones, nothing seems to resolve the issue.

**Comment by Gaurav-IcodeLabs** [comment]
same error

**Comment by wahabsamii** [comment]
Not fix

**Comment by Saurav-Imcrinox** [comment]
add this lines in proguard-rules file 

# Stripe SDK Push Provisioning - Keep required classes
-keep class com.stripe.android.pushProvisioning.** { *; }
-dontwarn com.stripe.android.pushProvisioning.**

# For reflection used inside Stripe SDK
-keepclassmembers class * {
    @com.stripe.android.core.injection.Injectable *;
}

-keepclassmembers class com.stripe.android.** {
    public *;
}


make sure you are using the compatible version of kotlin with react native in my case it was 

react native :- 0.75.2 
kotlinVersion = "1.9.22"
stripe version is also causing the issues so install specific versions 
    "@stripe/stripe-react-native": "^0.39.0",

**Comment by shafqatbari** [comment]
Facing this on

@stripe/stripe-react-native = 0.50.0
react-native = 0.80.0,

app/build.gradle
  kotlinVersion = "2.1.20"

**Comment by aryadaulat** [comment]
> Facing this on
> 
> @stripe/stripe-react-native = 0.50.0 react-native = 0.80.0,
> 
> app/build.gradle kotlinVersion = "2.1.20"

yeah I got same issue , a few days ago running well but error now

**Comment by N1k0l1n** [comment]
#2008   Somehow i fixed for : @stripe/stripe-react-native@0.38.6 & react-native": "0.80.1" , that was the most downloaded version. Other things you need to do for ext {
buildToolsVersion = "35.0.0"
minSdkVersion = 24
compileSdkVersion = 35
targetSdkVersion = 35
ndkVersion = "27.1.12297006"
kotlinVersion = "2.1.20"
}

at the end of the build.gradle after
apply plugin: "com.facebook.react.rootproject"

i have added :

subprojects { subProj ->
subProj.plugins.withId("com.android.library") {
subProj.android {
lintOptions {
abortOnError false
disable "RestrictedApi"
}
}
}
}

and on androidmanifest i have added :      uses-permission
        android:name="android.permission.POST_NOTIFICATIONS"
        tools:ignore="ProtectedPermissions"

**Comment by aryadaulat** [comment]
> [#2008](https://github.com/stripe/stripe-react-native/issues/2008) Somehow i fixed for : @stripe/stripe-react-native@0.38.6 & react-native": "0.80.1" , that was the most downloaded version. Other things you need to do for ext { buildToolsVersion = "35.0.0" minSdkVersion = 24 compileSdkVersion = 35 targetSdkVersion = 35 ndkVersion = "27.1.12297006" kotlinVersion = "2.1.20" }
> 
> at the end of the build.gradle after apply plugin: "com.facebook.react.rootproject"
> 
> i have added :
> 
> subprojects { subProj -> subProj.plugins.withId("com.android.library") { subProj.android { lintOptions { abortOnError false disable "RestrictedApi" } } } }
> 
> and on androidmanifest i have added : uses-permission android:name="android.permission.POST_NOTIFICATIONS" tools:ignore="ProtectedPermissions"

thanks , that's fix on me

**Comment by sanjivpaul** [comment]
I get also same issue

"@stripe/stripe-react-native": "^0.55.0",
"react-native": "0.76.0",

    ext {
        buildToolsVersion = "35.0.0"
        minSdkVersion = 24
        compileSdkVersion = 35
        targetSdkVersion = 34
        ndkVersion = "26.1.10909125"
        // kotlinVersion = "1.7.20" //"1.9.24"
        kotlin_version = "1.7.20"
        googlePlayServicesAuthVersion = "20.7.0" // google sign in
    }

these are my configurations 

and my code is FAILED in this error

> Task :stripe_stripe-react-native:compileDebugKotlin FAILED

how can i fix that?.
